internal/util/projutil/project_util.go

```diff
@@ -146,7 +155,47 @@ func ExecCmd(cmd *exec.Cmd) error {
 	cmd.Stderr = os.Stderr
 	err := cmd.Run()
 	if err != nil {
-		return fmt.Errorf("failed to exec %#v: %v", cmd.Args, err)
+		return fmt.Errorf("failed to exec %s %#v: %v", cmd.Path, cmd.Args, err)
+	}
+	return nil
+}
+
+// IsDockerMultistage checks whether the docker daemon is version >= 17.05.0.
+// Both ARG before FROM and multistage builds exist in docker version >= 17.05.0.
+func IsDockerMultistage() bool {
+	client, err := docker.NewEnvClient()
+	if err != nil {
+		log.Fatalf("get docker client: (%v)", err)
+	}
+	ver, err := client.ServerVersion(context.TODO())
+	if err != nil {
+		log.Fatalf("get docker version: (%v)", err)
+	}
+	dockerVer := semver.New(ver.Version)
+	reqVer := semver.New("17.05.0")
+	return !dockerVer.LessThan(*reqVer)
+}
+
+// IsDockerfileMultistage checks if dockerfile uses a multistage pipeline.
+func IsDockerfileMultistage(dockerfile string) bool {
+	df, err := ioutil.ReadFile(dockerfile)
+	if err != nil {
+		log.Fatal("Dockerfile not found")
+	}
+	return bytes.Count(df, []byte("FROM")) > 1
```

Comment:
It's not unusual for people to comment out a FROM line and temporarily use a different tag or repo, for testing, development, etc.. So you'd end up with a Dockerfile that looks like:

```
# FROM quay.io/acme/hotapp:1.6.2
FROM quay.io/mhrivnak/hotapp:latest
...
```

This function would detect that as a multi-stage Dockerfile. You could avoid that by using a regex that only matches `FROM` statements appearing at the beginning of a line.